integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/iterate/IterateSequentialFaultSequenceTestCase.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.esb.mediator.test.iterate;
+
+import org.apache.axis2.AxisFault;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.esb.integration.common.utils.ESBIntegrationTest;
+
+import static org.testng.Assert.assertEquals;
+
+/**
+ * Test case for https://wso2.org/jira/browse/ESBJAVA-3675
+ * Test whether a property set inside iterate mediator(sequential=true) can be retrieved inside fault sequence.
+ *
+ * Class mediator jar : ClassMediatorDemo-1.0.0.jar
+ * Class mediator name : com.wso2.example.CsvValidatorMediator
+ */
+
+public class IterateSequentialFaultSequenceTestCase extends ESBIntegrationTest {
+
+    @BeforeClass(alwaysRun = true)
+    public void setEnvironment() throws Exception {
+        super.init();
+        verifyProxyServiceExistence("iterateMediatorSequentialFaultSequenceTestProxy");
+    }
+
+    @Test(groups = "wso2.esb", description = "Tests property retrieval in fault sequence")
+    public void testPropertyRetrievalInFaultSequence() throws Exception {
+        try {
+            axis2Client.sendSimpleStockQuoteRequest(getProxyServiceURLHttp(
+                    "iterateMediatorSequentialFaultSequenceTestProxy"), null, "WSO2");
```

Comment:
Add assert fail after service invocation since you are expecting a exception. otherwise test would pass if error in not thrown.

Reply:
Done